libdrgn/dwarf_info.c

```diff
@@ -6563,19 +6577,22 @@ drgn_function_type_from_dwarf(struct drgn_debug_info *dbinfo,
 			is_variadic = true;
 			break;
 		case DW_TAG_template_type_parameter:
-			err = parse_template_parameter(dbinfo, file, &child,
-						       drgn_dwarf_template_type_parameter_thunk_fn,
-						       &builder.template_builder);
-			if (err)
-				goto err;
-			break;
 		case DW_TAG_template_value_parameter:
 			err = parse_template_parameter(dbinfo, file, &child,
-						       drgn_dwarf_template_value_parameter_thunk_fn,
 						       &builder.template_builder);
 			if (err)
 				goto err;
 			break;
+		case DW_TAG_GNU_template_parameter_pack:
+			r = dwarf_child(&child, &child_param);
+			while (r == 0) {
+				err = parse_template_parameter(dbinfo, file, &child_param,
+							       &builder.template_builder);
+				if (err)
+					goto err;
+				r = dwarf_siblingof(&child_param, &child_param);
+			}
+			break;
```

Comment:
This (and the other copy) are missing error handling for `dwarf_child()`/`dwarf_siblingof()`. It's probably easier to deal with by refactoring this into its own function at this point. Something like:

```c
static struct drgn_error *
drgn_parse_template_parameter_pack(struct drgn_debug_info *dbinfo,
				   struct drgn_elf_file *file, Dwarf_Die *die,
				   struct drgn_template_parameters_builder *builder)
{
	struct drgn_error *err;
	Dwarf_Child child;
	int r = dwarf_child(die, &child);
	while (r == 0) {
		err = parse_template_parameter(dbinfo, file, &child, builder);
		if (err)
			return err;
		r = dwarf_siblingof(&child, &child);
	}
	if (r == -1) {
		return drgn_error_create(DRGN_ERROR_OTHER,
					 "libdw could not parse DIE children");
	}
	return NULL;
}
```